
services:
  vnode-0:
    container_name: mechaind-validator-0
    image: "zkmelabs/mechain"
    networks:
      - mechain-network
    ports:
      - "28750:28750"
      - "27750:27750"
      - "9090:9090"
      - "1317:1317"
      - "26657:26657"
      - "8545:8545"
      - "8546:8546"
    volumes:
      - "./deployment/dockerup/.local/validator0:/app:Z"
    command: >
      /usr/bin/mechaind start --home /app
      --keyring-backend test
      --api.enabled-unsafe-cors true
      --address 0.0.0.0:28750
      --grpc.address 0.0.0.0:9090
      --p2p.laddr tcp://0.0.0.0:27750
      --p2p.external-address 0.0.0.0:27750
      --rpc.laddr tcp://0.0.0.0:26657
      --rpc.unsafe true
      --log_format json
  vnode-1:
    container_name: mechaind-validator-1
    image: "zkmelabs/mechain"
    networks:
      - mechain-network
    ports:
      - "28751:28750"
      - "27751:27750"
      - "9091:9090"
      - "1318:1317"
      - "26658:26657"
      - "8547:8545"
      - "8548:8546"
    volumes:
      - "./deployment/dockerup/.local/validator1:/app:Z"
    command: >
      /usr/bin/mechaind start --home /app
      --keyring-backend test
      --api.enabled-unsafe-cors true
      --address 0.0.0.0:28750
      --grpc.address 0.0.0.0:9090
      --p2p.laddr tcp://0.0.0.0:27750
      --p2p.external-address 0.0.0.0:27750
      --rpc.laddr tcp://0.0.0.0:26657
      --rpc.unsafe true
      --log_format json
  vnode-2:
    container_name: mechaind-validator-2
    image: "zkmelabs/mechain"
    networks:
      - mechain-network
    ports:
      - "28752:28750"
      - "27752:27750"
      - "9092:9090"
      - "1319:1317"
      - "26659:26657"
      - "8549:8545"
      - "8550:8546"
    volumes:
      - "./deployment/dockerup/.local/validator2:/app:Z"
    command: >
      /usr/bin/mechaind start --home /app
      --keyring-backend test
      --api.enabled-unsafe-cors true
      --address 0.0.0.0:28750
      --grpc.address 0.0.0.0:9090
      --p2p.laddr tcp://0.0.0.0:27750
      --p2p.external-address 0.0.0.0:27750
      --rpc.laddr tcp://0.0.0.0:26657
      --rpc.unsafe true
      --log_format json
  vnode-3:
    container_name: mechaind-validator-3
    image: "zkmelabs/mechain"
    networks:
      - mechain-network
    ports:
      - "28753:28750"
      - "27753:27750"
      - "9093:9090"
      - "1320:1317"
      - "26660:26657"
      - "8551:8545"
      - "8552:8546"
    volumes:
      - "./deployment/dockerup/.local/validator3:/app:Z"
    command: >
      /usr/bin/mechaind start --home /app
      --keyring-backend test
      --api.enabled-unsafe-cors true
      --address 0.0.0.0:28750
      --grpc.address 0.0.0.0:9090
      --p2p.laddr tcp://0.0.0.0:27750
      --p2p.external-address 0.0.0.0:27750
      --rpc.laddr tcp://0.0.0.0:26657
      --rpc.unsafe true
      --log_format json
networks:
  mechain-network:
    external: true
